core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -0,0 +1,330 @@
+    throw new UnsupportedOperationException();
+  }
+
+  @Deprecated
+  @Override
+  public void setTimeOut(int timeOut) {
+    if (timeOut == Integer.MAX_VALUE)
+      setTimeout(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+    else
+      setTimeout(timeOut, TimeUnit.SECONDS);
+  }
+
+  @Deprecated
+  @Override
+  public int getTimeOut() {
+    long timeout = getTimeout(TimeUnit.SECONDS);
+    if (timeout >= Integer.MAX_VALUE)
+      return Integer.MAX_VALUE;
+    return (int) timeout;
+  }
+
+  @Override
+  public void setRange(Range range) {
+    this.range = range;
+  }
+
+  @Override
+  public Range getRange() {
+    return range;
+  }
+
+  @Override
+  public void setBatchSize(int size) {
+    this.batchSize = size;
+  }
+
+  @Override
+  public int getBatchSize() {
+    return batchSize;
+  }
+
+  @Override
+  public void enableIsolation() {}
+
+  @Override
+  public void disableIsolation() {}
+
+  @Override
+  public synchronized void setReadaheadThreshold(long batches) {
+    Preconditions.checkArgument(batches > 0);
+    readaheadThreshold = batches;
+  }
+
+  @Override
+  public synchronized long getReadaheadThreshold() {
+    return readaheadThreshold;
+  }
+
+  @Override
+  public Authorizations getAuthorizations() {
+    return opts.auths;
+  }
+
+  @Override
+  public void addScanIterator(IteratorSetting cfg) {
+    super.addScanIterator(cfg);
+  }
+
+  @Override
+  public void removeScanIterator(String iteratorName) {
+    super.removeScanIterator(iteratorName);
+  }
+
+  @Override
+  public void updateScanIteratorOption(String iteratorName, String key, String value) {
+    super.updateScanIteratorOption(iteratorName, key, value);
+  }
+
+  private static final Range EMPTY_RANGE = new Range();
+
+  private class IterEnv extends BaseIteratorEnvironment {
+    @Override
+    public IteratorScope getIteratorScope() {
+      return IteratorScope.scan;
+    }
+
+    @Override
+    public boolean isFullMajorCompaction() {
+      return false;
+    }
+
+    @Override
+    public Authorizations getAuthorizations() {
+      return opts.auths;
+    }
+
+    @Override
+    public boolean isSamplingEnabled() {
+      return RFileScanner.this.getSamplerConfiguration() != null;
+    }
+
+    @Override
+    public SamplerConfiguration getSamplerConfiguration() {
+      return RFileScanner.this.getSamplerConfiguration();
+    }
+  }
+
+  @Override
+  public Iterator<Entry<Key,Value>> iterator() {
+    try {
+      RFileSource[] sources = opts.in.getSources();
+      List<SortedKeyValueIterator<Key,Value>> readers = new ArrayList<>(sources.length);
+      for (int i = 0; i < sources.length; i++) {
+        FSDataInputStream inputStream = (FSDataInputStream) sources[i].getInputStream();
+        readers.add(new RFile.Reader(new CachableBlockFile.Reader(inputStream, sources[i].getLength(), opts.in.getConf(), dataCache, indexCache,
+            AccumuloConfiguration.getDefaultConfiguration())));
```

Comment:
Not sure, I haven't looked closely enough. I was just noting that `AccumuloConfiguration.getDefaultConfiguration()` is creating `sources.length` instances of `AccumuloConfiguration`. It's not returning an immutable singleton.